adal/src/telemetry/java/com/microsoft/aad/adal/HttpEvent.java

```diff
@@ -0,0 +1,113 @@
+import java.util.Map;
+
+class HttpEvent extends DefaultEvent {
+    HttpEvent(final String eventName) {
+        getEventList().add(Pair.create(EventStrings.EVENT_NAME, eventName));
+    }
+
+    void setUserAgent(final String userAgent) {
+        setEvent(EventStrings.HTTP_USER_AGENT, userAgent);
+    }
+
+    void setMethod(final String method) {
+        setEvent(EventStrings.HTTP_METHOD, method);
+    }
+
+    void setQueryParameters(final String queryParameters) {
+        setEvent(EventStrings.HTTP_QUERY_PARAMETERS, queryParameters);
+    }
+
+    void setResponseCode(final Integer responseCode) {
+        setEvent(EventStrings.HTTP_RESPONSE_CODE, responseCode.toString());
+    }
+
+    void setApiVersion(final String apiVersion) {
+        setEvent(EventStrings.HTTP_API_VERSION, apiVersion);
+    }
+
+    void setHttpPath(final URL httpPath) {
+        final String authority = httpPath.getAuthority();
+        final Discovery discovery = new Discovery();
+        if (!discovery.getValidHosts().contains(authority)) {
+            return;
+        }
+
+        final String[] splitArray = httpPath.getPath().split("/");
+
+        final StringBuilder logPath = new StringBuilder();
+        logPath.append(httpPath.getProtocol());
+        logPath.append("://");
+        logPath.append(authority);
+        logPath.append("/");
+
+        // we do not want to send tenant information
+        // index 0 is blank
+        // index 1 is tenant
+        for (int i = 2; i < splitArray.length; i++) {
+            logPath.append(splitArray[i]);
+            logPath.append("/");
+        }
+        setEvent(EventStrings.HTTP_PATH, logPath.toString());
+    }
+
+    void setOauthErrorCode(final String errorCode) {
+        setEvent(EventStrings.OAUTH_ERROR_CODE, errorCode);
+    }
+
+    void setRequestIdHeader(final String requestIdHeader) {
+        setEvent(EventStrings.REQUEST_ID_HEADER, requestIdHeader);
+    }
+
+    @Override
+    public void processEvent(final Map<String, String> dispatchMap) {
+        final Object countObject = dispatchMap.get(EventStrings.HTTP_EVENT_COUNT);
+
+        if (countObject == null) {
+            dispatchMap.put(EventStrings.HTTP_EVENT_COUNT, "1");
+        } else {
+            dispatchMap.put(EventStrings.HTTP_EVENT_COUNT,
+                    Integer.toString(Integer.parseInt((String) countObject) + 1));
+        }
+
+        final List eventList = getEventList();
+        final int size = eventList.size();
+        for (int i = 0; i < size; i++) {
+            final Pair eventPair = (Pair<String, String>) eventList.get(i);
+            final String name = (String) eventPair.first;
+
+            if (name.equals(EventStrings.HTTP_RESPONSE_CODE) || name.equals(EventStrings.REQUEST_ID_HEADER)
+                    || name.equals(EventStrings.OAUTH_ERROR_CODE) || name.equals(EventStrings.HTTP_PATH)) {
+                dispatchMap.put(name, (String) eventPair.second);
+            }
+        }
```

Comment:
maybe the next release. do you have any ideas?

---
In reply to: [90371438](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/736#discussion_r90371438) [](ancestors = 90371438)